packages/ember-routing/lib/system/route.js

```diff
@@ -605,23 +605,19 @@ var Route = EmberObject.extend(ActionHandler, Evented, {
   /**
     The `loading` action is fired on the route when a route's `model`
     hook returns a promise that is not already resolved. The current
-    `Transition` object is the first parameter and the route that
+    `transition` object is the first parameter and the route that
     triggered the loading event is the second parameter.
 
     ```javascript
     App.ApplicationRoute = Ember.Route.extend({
       actions: {
         loading: function(transition, route) {
-          var view = Ember.View.create({
-            classNames: ['app-loading']
-          })
-          .append();
+          let controller = this.controllerFor('foo');
+          controller.set('currentlyLoading', true);
 
-          this.router.one('didTransition', function() {
-            view.destroy();
+          transition.promise.finally(function() {
```

Comment:
It is, but as DOC says, it is there if you want to pass around a promise without passing the full transition. In this case,  I think [`transition.finally`](https://github.com/tildeio/router.js/blob/master/lib/router/transition.js#L167-L182) would do too.